source/class/qx/data/Array.js

```diff
@@ -1017,7 +1017,7 @@ qx.Class.define("qx.data.Array", {
      *   <li><code>index</code>: the index of the current item</li>
      *   <li><code>array</code>: The native array instance, NOT the data array instance.</li>
      * @param self {var?undefined} The context of the callback.
-     * @return {Boolean} <code>true</code>, if any element passed the test function.
+     * @return {var | undefined} The found item.
```

Comment:
We don't tend to use `var`, and optional results are indicated with a trailing `?` - I think that the correct jsdoc would be `@return {*?}` or `@return {Object?}` - but not sure which is correct/better

Reply:
Agree, but nevertheless it is still much better than Boolean and the wrong description of return which confuses a documentation reader. In other methods of qx.data.Array I noticed there is used for example for removeAt method. I doubt that Object? is valid jsdoc. Especially an undefined value is needed to be highlighted/emphasized.

Reply:
We use `?` to indicate optional parameters. I've never seen, and would find confusing, a return value indicated that way. It's not optional; it can just be of various types (including undefined).

We use `var` at multiple places in the framework, e.g., in qx.bom.Event and qx.core.MLogging. I think the PR's description of the `{var | undefined}` is exactly appropriate. (We have discussed possibly moving to `*` particularly as it relates to arrays (I think) but I don't know that we ever came to a conclusion on that or decided to begin to use it. If we want to do so, this return value could be documented as `{* | undefined}`.

Regarding changing `undefined` to `null`... See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find. `undefined` is correct.

Reply:
@derrell I am not sure about "| undefined" because other typical methods which can return undefined/null value have just "var" in JSDoc. But another thing I've noticed in ApiViewer in min/max methods. JSDoc of these methods have Number | null type but in ApiView they are shown as | null. Is it bug of ApiViewer or not correct notation?

Reply:
That sounds like a bug in ApiViewer. Can you maybe provide an example class documented that way, but shows up incorrectly? (A new PR with that would be better than here.)

Reply:
@derrell I'll do it probably like issue for **ApiViewer**. The class is the same one:  `qx.data.Array`. Last detail I would like to clarify  `var | undefined` is fine, no need to change onto just `var` ?

Reply:
https://github.com/qooxdoo/qxl.apiviewer/issues/48